connect_ext_test.go

```diff
@@ -699,6 +744,42 @@ func TestBidiRequiresHTTP2(t *testing.T) {
 	)
 }
 
+func TestCompressMinBytesClient(t *testing.T) {
+	t.Parallel()
+	serve := func() (pingv1connect.PingServiceClient, *string) {
+		mux := http.NewServeMux()
+		var contentEncoding string
+		go mux.Handle("/", http.HandlerFunc(func(writer http.ResponseWriter, request *http.Request) {
+			contentEncoding = request.Header.Get("Content-Encoding")
```

Comment:
nit: This feels a bit racy to use a `*string` here and rely on the handler being called (without using sync.WaitGroup or other things to ensure things have completed).

Once option might to be to return a function which takes a PingRequest (or a string) and returns a PingResponse and content encoding. Then you can ensure the concurrency is sane and potentially also minimize duplication in the tests below.